common/src/main/java/org/keycloak/common/constants/KerberosConstants.java

```diff
@@ -71,6 +71,11 @@ public class KerberosConstants {
     public static final String SERVER_PRINCIPAL = "serverPrincipal";
     public static final String KEYTAB = "keyTab";
     public static final String DEBUG = "debug";
+    public static final String KERBEROS_PRINCIPAL_ATTRIBUTE = "krbPrincipalAttribute";
+    public static final String KERBEROS_PRINCIPAL_LDAP_ATTRIBUTE_KRB5_PRINCIPAL_NAME = "krb5PrincipalName"; // Used for instance in ApacheDS
```

Comment:
@mposolda Isn't possible to infer the LDAP attribute name based on the vendor? Wondering if it helps to avoid forcing admins to save typing when creating the provider for well-known vendors.

Reply:
Yes, that's good idea and I've already did that at the admin console level. Admin console "suggests" the attribute based on the vendor, which is selected by administrator.

But now I've updated it to also add the attribute on server-side when creating new LDAP provider, so that it is applied also for the cases when LDAP provider is created directly by REST API for example. That's what we're doing for some other LDAP settings as well.